doc/vlan/switchport-mode-support/Switchport Mode and VLAN CLI Enhancement.md

```diff
@@ -0,0 +1,760 @@
+
+We have given examples/usage for switchport mode configuration from routed to access on physical port. This works in the same way for PortChannels as they do for physical ports.
+
+#### Configuring Port & PortChannel from Routed to Trunk 
+
+* Physical Port Configuration
+
+In these examples, Ethernet4 will be configured as “Trunk” from “Routed” on Port.
+
+By default, all ports are in routed mode and have IP assigned. For Switchport configuration we have to remove IP assignment otherwise CLI will show following Error:
+
+**For Ethernet4**
+
+<img src="https://user-images.githubusercontent.com/61490193/236170350-be5d23a1-ba77-4a1e-8cfb-5b9b795c48a9.png" width="65%" height="30%">
+
+
+Following Steps will be taken to configure “Trunk” from “Routed” on a Port
+
+**1.** IP Removal on  Ethernet4
+
+```
+admin@sonic:~$ sudo config interface ip remove Ethernet4 10.0.0.2/31
+```
+
+**2.** Configure Ethernet4 as Trunk
+
+```
+admin@sonic:~$ sudo config switchport mode Trunk Ethernet4       
+```
+<img src="https://user-images.githubusercontent.com/61490193/236542261-682ebb20-07bd-4a34-ab9b-dc92bfc95bf0.png" width="65%" height="30%">
+**3.** Untagged Vlan Member Assignment on Trunk Port
+
+```
+admin@sonic:~$ sudo config vlan member add 3 -u Ethernet4          
+```
+
+<img src="https://user-images.githubusercontent.com/61490193/236294289-44cbceff-a3be-4460-a5c0-a8f219720466.png" width="65%" height="30%">
+
+**4.** Multiple Untagged Vlan Member Assignment on Trunk Port
+Ethernet4 is in Trunk mode, it can have only 1 untagged member. Configuring More than 1 untagged member on trunk Port will show following error:
+
+
+<img src="https://user-images.githubusercontent.com/61490193/236170542-587403ed-3cf0-465a-9e58-fff7c8130508.png" width="65%" height="30%">
+
+
+**5.** Tagged Vlan Member Assignment on Trunk Port
+
+Ethernet4 is in trunk mode, it can have multiple tagged members. Configuring tagged member on trunk Port will show following
+
+
+<img src="https://user-images.githubusercontent.com/61490193/236294274-61705636-4f9f-4756-9062-58d55334b836.png" width="65%" height="30%">
+
+**6.** IP Assignment on Trunk Port
+
+
+Ethernet4 is in Trunk mode, IP assignment on the Trunk port is not allowed. Configuring IP Assignment on Trunk Port will show following error:
+
+<img src="https://user-images.githubusercontent.com/61490193/236170307-9c58b1c3-4ad8-4061-9908-4163a833a315.png" width="65%" height="40%">
+ 
+**7.** Change Mode from Trunk to Routed 
+
+Ethernet4 is in Trunk mode, Changing Trunk port to routed is not possible until it has an untagged and tagged members. Changing Trunk to routed will show following error:
+
+
+<img src="https://user-images.githubusercontent.com/61490193/236533178-bba78b3f-f429-4042-8de9-0decbe938f36.png" width="65%" height="40%">
+ 
+**8.** Change Mode from Trunk to Access
+
+Ethernet4 is in Trunk mode, Changing Trunk port to access is possible and  its  untagged members wll retain. Changing Trunk to access will show following:
```

Comment:
What about from Access to Trunk does it also retain the untagged vlan membership?